google/cloud/functions/CMakeLists.txt

```diff
@@ -80,7 +81,8 @@ target_include_directories(functions_framework_cpp SYSTEM
                            PUBLIC $<INSTALL_INTERFACE:include>)
 target_link_libraries(
     functions_framework_cpp PUBLIC absl::time Boost::headers
-                                   Boost::program_options Threads::Threads)
+                                   Boost::program_options Threads::Threads
+                                   nlohmann_json::nlohmann_json)
```

Comment:
I wonder if making `nlohmann_json::nlohmann_json` a `PRIVATE` requirement would be better.  As a header-only library that is only used in `.cc` files, there is no need for downstream targets to need that target.

We could also fix the build problems if we add:

```cmake
find_dependency(nlohmann_json)
```

around here:

https://github.com/GoogleCloudPlatform/functions-framework-cpp/blob/e79a2125abdae3ef66b83f55fa854e18f1bfb926/google/cloud/functions/config.cmake.in#L16-L19

But I think the `PRIVATE` thing would work better.  Thoughts?

Reply:
Yes, `PRIVATE` fits better in this case, well observed. Let me update it, thank you for reviewing.

Reply:
Done, it's using PRIVATE now.